api/web_conn.go

```diff
@@ -118,6 +118,10 @@ func (c *WebConn) InvalidateCache() {
 	c.hasPermissionsToTeam = make(map[string]bool)
 }
 
+func (c *WebConn) InvalidateCacheForChannel(channelId string) {
```

Comment:
Is another method really needed?  can't we just reduce the complexity and call `InvalidateCacheForUser()` since it clear's all the users cache?

Leaving a channel is so rare invalidating all the users cache isn't a big hit.